clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/gitRepo/GitRepoArtifactCredentials.java

```diff
@@ -57,24 +61,62 @@ public String getType() {
 
   @Override
   public InputStream download(Artifact artifact) throws IOException {
-    String repoReference = artifact.getReference();
+    String repoUrl = artifact.getReference();
     String subPath = artifactSubPath(artifact);
-    String remoteRef = artifactVersion(artifact);
-    Path stagingPath =
-        Paths.get(System.getProperty("java.io.tmpdir"), UUID.randomUUID().toString());
-    String repoBasename = getRepoBasename(repoReference);
+    String branch = artifactVersion(artifact);
+    Path stagingPath = gitRepoFileSystem.getLocalClonePath(repoUrl, branch);
+    String repoBasename = getRepoBasename(repoUrl);
     Path outputFile = Paths.get(stagingPath.toString(), repoBasename + ".tgz");
 
-    // delete temporary files before returning
-    try (Closeable ignored = () -> FileUtils.deleteDirectory(stagingPath.toFile())) {
-      executor.clone(repoReference, remoteRef, stagingPath);
-
-      log.info("Creating archive for git/repo {}", repoReference);
-
-      executor.archive(
-          Paths.get(stagingPath.toString(), repoBasename), remoteRef, subPath, outputFile);
+    if (!gitRepoFileSystem.canRetainClone()) {
+      // delete clone before returning
+      try (Closeable ignored = () -> FileUtils.deleteDirectory(stagingPath.toFile())) {
+        return getInputStream(repoUrl, subPath, branch, stagingPath, repoBasename, outputFile);
+      }
+    } else {
+      // clones are deleted by gitRepoFileSystem depending on retention period
+      return getInputStream(repoUrl, subPath, branch, stagingPath, repoBasename, outputFile);
+    }
+  }
 
-      return new FileInputStream(outputFile.toFile());
+  @NotNull
+  private FileInputStream getInputStream(
+      String repoUrl,
+      String subPath,
+      String branch,
+      Path stagingPath,
+      String repoBasename,
+      Path outputFile)
+      throws IOException {
+    try {
+      if (gitRepoFileSystem.tryLock(repoUrl, branch, MAX_WAIT_LOCK_SEC, TimeUnit.SECONDS)) {
```

Comment:
Isn't 10s too short? We've seen some checkout take minutes. Can we make it user configurable?